src/settings.c

```diff
@@ -303,10 +312,95 @@ static void set_key_binding(int *key, const char **bind)
     }
 }
 
-int settings_load(struct user_settings *s, const char *patharg)
+static bool get_settings_path(char *path, unsigned int path_size, const char *patharg)
+{
+    if (patharg != NULL) {
+        snprintf(path, path_size, "%s", patharg);
+        return true;
+    }
+
+    char *user_config_dir = get_user_config_dir();
+    snprintf(path, path_size, "%s%stoxic.conf", user_config_dir, CONFIGDIR);
+    free(user_config_dir);
+
+    /* make sure path exists or is created on first time running */
+    if (!file_exists(path)) {
+        FILE *fp = fopen(path, "w");
+
+        if (fp == NULL) {
+            fprintf(stderr, "failed to open config path: %s\n", path);
+            return false;
+        }
+
+        fclose(fp);
+    }
+
+    return true;
+}
+
+int settings_load_friends(const char *patharg)
+{
+    config_t cfg[1];
+    const char *str = NULL;
+
+    config_init(cfg);
+
+    char path[MAX_STR_SIZE] = {0};
+
+    if (!get_settings_path(path, sizeof(path), patharg)) {
+        config_destroy(cfg);
+        return -1;
+    }
+
+    if (!config_read_file(cfg, path)) {
+        fprintf(stderr, "settings read error: %s:%d - %s\n", config_error_file(cfg),
+                config_error_line(cfg), config_error_text(cfg));
+        config_destroy(cfg);
+        return -2;
+    }
+
+    const config_setting_t *setting = config_lookup(cfg, friend_strings.self);
+
+    if (setting == NULL) {
+        config_destroy(cfg);
+        return -3 ;
+    }
+
+    const int num_friends = config_setting_length(setting);
+
+    for (int i = 0; i < num_friends; ++i) {
+        const config_setting_t *keys = config_setting_get_elem(setting, i);
+
+        if (keys == NULL) {
+            continue;
+        }
+
+        const char *public_key = config_setting_name(keys);
+
+        if (public_key == NULL) {
+            continue;
+        }
+
+        if (strlen(public_key) != (TOX_PUBLIC_KEY_SIZE * 2) + 3) {
```

Comment:
where do the `+3` come frome?

Reply:
looks like config values for public keys start with "pk_". Either document and/or add a "pk_" prefix check to make it explicit.

Reply:
On second thought, I guess we should do an explicit check to prevent any potential undocumented behaviour.